frontend/src/CollectionForm/CollectionForm.tsx

```diff
@@ -73,47 +77,60 @@ function CollectionForm(props: ICollectionFormProps) {
     if (!props.collection || !Object.keys(props.collection)) return;
     setTitle(props.collection.name);
     setDescription(props.collection.description);
-  }, [props.collection])
+  }, [props.collection]);
 
   useEffect(() => {
     if (!props.collection || murals.length === 0) return;
     let temp: any[] = [];
     props.collection.murals?.forEach((tourMural: any) => {
       let found = murals.find((mural: any) => mural.id === tourMural.id);
-      temp = ([...temp, found]);
-    })
+      temp = [...temp, found];
+    });
     setMuralsInCollection(temp);
   }, [props.collection, murals]);
 
   /**
    * Enable editable form fields for admin users
    */
-  const userContext: any = useContext(Context)
+  const userContext: any = useContext(Context);
   useEffect(() => setIsAdmin(!!(userContext as any).user), [userContext]);
 
   const handleAddMural = (addedMural: any) => {
-    if (!addedMural) return
-    if (muralsInCollection.some((currMural: any) => currMural.id === addedMural.id))
-      return
+    if (!addedMural) return;
+    if (
+      muralsInCollection.some(
+        (currMural: any) => currMural.id === addedMural.id
+      )
+    )
+      return;
 
     setMuralsInCollection((oldMurals: any) => [...oldMurals, addedMural]);
   };
 
   const handleSave = () => {
     if (!title.length || !description.length) return;
 
-    axios.post(GET_ALL_COLLECTION, {
-      collection: {
-        name: title,
-        description: description
-      },
-      murals: muralsInCollection.map((mural: any) => mural.id)
-    },
-    {
-      headers: {
-        authorization: userContext.token.i
-      }
-    })
+    if (!userContext.token) {
+      alert("Please log in again.");
+      return;
+    }
+
+    axios
+      .post(
+        GET_ALL_COLLECTION,
+        {
+          collection: {
+            name: title,
+            description: description,
+          },
+          murals: muralsInCollection.map((mural: any) => mural.id),
+        },
+        {
+          headers: {
+            authorization: userContext.token,
```

Comment:
only real change here and in `<TourForm />`: use `userContext.token` instead of `userContext.token.i`